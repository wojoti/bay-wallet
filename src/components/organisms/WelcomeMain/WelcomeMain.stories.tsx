import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import WelcomeMain from './WelcomeMain';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Organisms/WelcomeMain',
  component: WelcomeMain,
  argTypes: {
    onButtonPress: {action: 'onButtonPress'},
  },
} as ComponentMeta<typeof WelcomeMain>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof WelcomeMain> = args => (
  <WelcomeMain {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
};
