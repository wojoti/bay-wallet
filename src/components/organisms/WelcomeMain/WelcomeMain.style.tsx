import {StyleSheet} from 'react-native';

interface Props {
  mainColor?: string;
}
const style = ({mainColor}: Props) =>
  StyleSheet.create({
    wrapper: {
      backgroundColor: mainColor,
    },
    content: {
      width: '85%',
    },
  });
export default style;
