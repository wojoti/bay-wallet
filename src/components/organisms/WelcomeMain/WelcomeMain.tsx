import {Button, Container, Header, Img, ImgType, Label} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './WelcomeMain.style';

export type WelcomeMainProps = PropsWithChildren<{
  onButtonPress: () => void;
  mainColor: string;
  accentColor: string;
  testId?: string;
}>;

const Welcomemain = ({
  onButtonPress,
  mainColor,
  accentColor,
  testId,
}: WelcomeMainProps) => {
  return (
    <Container
      flex={1}
      justify="center"
      align="center"
      customStyle={style({mainColor}).wrapper}
      testId={testId}>
      <Container flex={1} justify="flex-end">
        <Img
          src={ImgType.logoSplashColor}
          height={330}
          alt={''}
          testId={'welcomemain-img'}
        />
      </Container>
      <Container flex={1.1} align="center" customStyle={style({}).content}>
        <Container align="center" flex={2.6}>
          <Label color={accentColor} size={30} testId={'welcomemain-label'}>
            Welcome to,
          </Label>
          <Header color={accentColor} size={50} testId={'welcomemain-header'}>
            BayWallet
          </Header>
          <Label
            color={accentColor}
            size={14}
            align="center"
            testId={'welcomemain-description'}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore
          </Label>
        </Container>
        <Container flex={1}>
          <Button
            type="tertiary"
            onPress={onButtonPress}
            testId={'welcomemain-button'}>
            GET STARTED
          </Button>
        </Container>
      </Container>
    </Container>
  );
};

export default Welcomemain;
