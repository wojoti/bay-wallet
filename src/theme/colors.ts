export const lightPrimary = '#4E439E';
export const primary = '#363062';
export const secondary = 'white';

export const darkGrey = '#595959';
export const grey = '#6B6B6B';
export const vlightGrey = '#D1D1D1';
export const borderGrey = '#E2E2E2';

export const iconGrey = '#BDBDBD';

export const searchGrey = '#F5F5F5';

export const separator = '#F6F6F6';

export const green = '#61C277';
export const lime = '#53E520';
export const red = '#E52020';

export const successGreen = '#3EB290';

export const pink = '#D8B9C3';
export const darkPurple = '#26204E';
export const balanceWhite = '#F8F8FF';

export const marketDarkGrey = '#3D454B';

export const cardYellow = '#FFEF60';

export const buttonGrey = '#EFEFEF';
