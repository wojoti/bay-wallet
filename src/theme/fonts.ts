export const font = 'Poppins-Light';
export const boldFont = 'Poppins-Bold';
