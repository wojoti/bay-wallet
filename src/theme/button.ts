import {StyleSheet} from 'react-native';
import {primary, secondary} from './colors';
const buttonStyle = StyleSheet.create({
  primary: {
    borderWidth: 0,
    backgroundColor: primary,
  },
  secondary: {
    borderColor: primary,
    backgroundColor: secondary,
  },
});

export const buttonTextStyle = StyleSheet.create({
  primary: {
    color: secondary,
  },
  secondary: {
    color: primary,
  },
});

export default buttonStyle;
