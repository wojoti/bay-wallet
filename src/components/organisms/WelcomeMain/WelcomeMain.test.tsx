import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import {primary, secondary} from 'theme/colors';
import WelcomeMain, {WelcomeMainProps} from './WelcomeMain';

const handleClick = jest.fn();
const props: WelcomeMainProps = {
  onButtonPress: handleClick,
  mainColor: primary,
  accentColor: secondary,
  testId: 'test-welcomemain-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<WelcomeMain {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render welcomemain', () => {
  render(<WelcomeMain {...props} />);
  const welcomemainElement = screen.getByTestId('test-welcomemain-id');
  expect(welcomemainElement).toBeOnTheScreen();
});

test('should render welcomemain with img', () => {
  render(<WelcomeMain {...props} />);
  const welcomemainElement = screen.getByTestId('test-welcomemain-id');
  expect(welcomemainElement).toBeOnTheScreen();
  const welcomemainImgElement =
    within(welcomemainElement).getByTestId('welcomemain-img');
  expect(welcomemainImgElement).toBeOnTheScreen();
});

test('should render welcomemain with label - welcome', () => {
  render(<WelcomeMain {...props} />);
  const welcomemainElement = screen.getByTestId('test-welcomemain-id');
  expect(welcomemainElement).toBeOnTheScreen();
  const welcomemainLabelElement =
    within(welcomemainElement).getByTestId('welcomemain-label');
  expect(welcomemainLabelElement).toBeOnTheScreen();
  expect(welcomemainLabelElement).toHaveTextContent('Welcome to,');
});

test('should render welcomemain with header', () => {
  render(<WelcomeMain {...props} />);
  const welcomemainElement = screen.getByTestId('test-welcomemain-id');
  expect(welcomemainElement).toBeOnTheScreen();
  const welcomemainHeaderElement =
    within(welcomemainElement).getByTestId('welcomemain-header');
  expect(welcomemainHeaderElement).toBeOnTheScreen();
  expect(welcomemainHeaderElement).toHaveTextContent('BayWallet');
});

test('should render welcomemain with description', () => {
  render(<WelcomeMain {...props} />);
  const welcomemainElement = screen.getByTestId('test-welcomemain-id');
  expect(welcomemainElement).toBeOnTheScreen();
  const welcomemainLabelElement = within(welcomemainElement).getByTestId(
    'welcomemain-description',
  );
  expect(welcomemainLabelElement).toBeOnTheScreen();
  expect(welcomemainLabelElement).toHaveTextContent(
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore',
  );
});

test('should render welcomemain with button and correct button style', () => {
  render(<WelcomeMain {...props} />);
  const welcomemainElement = screen.getByTestId('test-welcomemain-id');
  expect(welcomemainElement).toBeOnTheScreen();
  const welcomemainButtonElement =
    within(welcomemainElement).getByTestId('welcomemain-button');
  expect(welcomemainButtonElement).toBeOnTheScreen();
  expect(welcomemainButtonElement).toHaveStyle({backgroundColor: secondary});
  const textElement = within(welcomemainButtonElement).getByTestId(
    'button-text',
  );
  expect(textElement).toBeOnTheScreen();
  expect(textElement).toHaveStyle({color: 'black'});
});

test('should render welcomemain with button - handle onPress', () => {
  render(<WelcomeMain {...props} />);
  const welcomemainElement = screen.getByTestId('test-welcomemain-id');
  expect(welcomemainElement).toBeOnTheScreen();
  const welcomemainButtonElement =
    within(welcomemainElement).getByTestId('welcomemain-button');
  expect(welcomemainButtonElement).toBeOnTheScreen();
  expect(handleClick).toBeCalledTimes(0);
  fireEvent.press(welcomemainButtonElement);
  expect(handleClick).toBeCalledTimes(1);
});
