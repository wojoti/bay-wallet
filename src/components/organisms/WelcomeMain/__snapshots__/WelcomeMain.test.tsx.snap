// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "100%",
      },
      {
        "backgroundColor": "#363062",
      },
    ]
  }
  testID="test-welcomemain-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <SvgMock
      height={330}
      testID="welcomemain-img"
      width="100%"
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1.1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "width": "85%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 2.6,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Poppins-Light",
            "fontSize": 30,
            "fontWeight": "400",
            "textAlign": "left",
          }
        }
        testID="welcomemain-label"
      >
        Welcome to,
      </Text>
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Poppins-Bold",
            "fontSize": 50,
            "letterSpacing": undefined,
            "textAlign": "left",
          }
        }
        testID="welcomemain-header"
      >
        BayWallet
      </Text>
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Poppins-Light",
            "fontSize": 14,
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
        testID="welcomemain-description"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "white",
              "borderWidth": 0,
            },
          ]
        }
        testID="welcomemain-button"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "black",
              },
            ]
          }
          testID="button-text"
        >
          GET STARTED
        </Text>
      </View>
    </View>
  </View>
</View>
`;
